app/models/attachment.rb

```diff
@@ -9,36 +9,47 @@ class Attachment < ApplicationRecord
   before_create :set_file_extension
 
   def push_event_data
-    data = {
+    return base_data.merge(location_data) if file_type.to_sym == :location
+    return base_data.merge(fallback_data) if file_type.to_sym == :fallback
+
+    base_data.merge(file_data)
+  end
+
+  private
+
+  def file_data
```

Comment:
@pranavrajs Couldn't come up with a better name for this method. What should we call this?

Reply:
`file_meta_data` ?

Reply:
Let me update it.